xprof/tally.hpp

```diff
@@ -425,43 +379,18 @@ auto get_uniq_tally(Map<std::tuple<K...>, V> &input) {
   return tuple_set;
 }
 
-//! Add the total values at the end of the table (represented as a vector of tuples)".
-/*!
-\param m vector of tuples.
-
-EXAMPLE:
-  input   vector{
-                pair{"zeModuleCreate"),  CoreTime},
-                pair{"zeModuleDestroy"), CoreTime},
-                pair{"zeMemFree"),       CoreTime}
-          }
-  output (update first param by reference)
-
-          vector{
-                pair{"zeModuleCreate"),  CoreTime},
-                pair{"zeModuleDestroy"), CoreTime},
-                pair{"zeMemFree"),       CoreTime},
-                pair{"Total"),           CoreTime}
-          }
-
-*/
 template <typename TC, typename = std::enable_if_t<std::is_base_of_v<TallyCoreBase, TC>>>
 void add_footer(std::vector<std::pair<thapi_function_name, TC>> &m) {
-
-  // Create the final Tally
   TC tot{};
-  for (auto const &keyval : m)
-    tot += keyval.second;
-  // Finalize (compute ratio)
-  for (auto &keyval : m)
-    keyval.second.finalize(tot);
+  for (auto const &[_, t] : m)
+    tot += t;
+  m.push_back( { "Total", tot } );
 
-  if (tot.error == tot.count)
-    tot.duration_ratio = NAN;
-
-  m.push_back(std::make_pair(std::string("Total"), tot));
+  for (auto &[_, t] : m)
+    t.compute_duration_ratio(tot);
```

Comment:
```suggestion
    keyval.second.compute_duration_ratio(tot);
```

Reply:
Why do you prefer `keyval.second` ?

Reply:
Good question, I like your approach with the dummy variable `_`, but we still have the compiler indicating this variable is unused.

Reply:
* g++ (SUSE Linux) 7.5.0
* gcc (SUSE Linux) 7.5.0

Reply:
Fixed with ef7f5d5